app/controllers/admin/dashboards_controller.rb

```diff
@@ -1,20 +1,16 @@
 class Admin::DashboardsController < Admin::BaseController
 
   def show
-    @users = User.charity_admins(current_user.charities.first)
     @charity = current_user.charities.first
+    @users   = User.charity_admins(current_user.charities.first)
 
-    if current_user.charity_admin?
-      charity_id = current_user.user_roles.first.charity_id
-      if !charity_id.nil?
-        @charity = Charity.find(charity_id)
-      end
-    end
     if current_user.platform_admin?
       @users = User.all_admins
+      @platform_admin    = current_user
       @pending_charities = Charity.pending_charities
-      @online_charities = Charity.online_charities
+      @online_charities  = Charity.online_charities
       @offline_charities = Charity.offline_charities
     end
   end
+
```

Comment:
Just so you know, current_user is a helper method from within the views, and it is not necessary to use it if all you're doing is defining an variable for the view from the controller to equal current_user.